app/services/group_stage_service.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class GroupStageService
+  def self.generate_group_stage(groups)
+    return false if groups.length.zero?
+
+    average_group_size = (groups.flatten.length.to_f / groups.length.to_f)
+    if (average_group_size % 1).zero?
+      groups = groups.map { |group| get_group_object_from(group) }
+      group_stage = Stage.new level: -1, groups: groups
```

Comment:
same as above.

Reply:
What? :see_no_evil:

Reply:
Again I think this improves readability

Reply:
but the word groups is already in the attribute key? so it's already clear what the result is supposed to be

Reply:
But I still think that it is nicer to read. The more there is within one line, the more you will have to think yourself while trying to understand it. Also when a change within that line occurs, the diff is easier to understand too, as you do not have to search as long to find the (minor) difference